crates/volta-core/src/toolchain/mod.rs

```diff
@@ -52,37 +52,38 @@ impl Toolchain {
                 file: path.to_owned(),
             })?;
 
-        let platform = serial::Platform::from_json(src)?.into_platform()?;
+        let platform = serial::Platform::from_json(src)?
+            .into_platform()?
+            .map(Rc::new);
         if platform.is_some() {
             debug!("Found default configuration at '{}'", path.display());
         }
         Ok(Toolchain { platform })
     }
 
-    pub fn platform_ref(&self) -> Option<&PlatformSpec> {
-        self.platform.as_ref()
+    pub fn platform(&self) -> Option<Rc<PlatformSpec>> {
+        self.platform.clone()
     }
 
     /// Set the active Node version in the default platform file.
-    pub fn set_active_node(&mut self, node_version: &NodeVersion) -> Fallible<()> {
+    pub fn set_active_node(&mut self, node_version: &Version) -> Fallible<()> {
         let mut dirty = false;
 
-        if let Some(ref mut platform) = self.platform {
-            if platform.node_runtime != node_version.runtime {
-                platform.node_runtime = node_version.runtime.clone();
-                dirty = true;
-            }
-
-            if platform.npm.as_ref() != Some(&node_version.npm) {
-                platform.npm = Some(node_version.npm.clone());
+        if let Some(platform) = &self.platform {
+            if platform.node.version != *node_version {
+                self.platform = Some(Rc::new(PlatformSpec {
+                    node: SourcedVersion::default(node_version.clone()),
+                    npm: platform.npm.clone(),
+                    yarn: platform.yarn.clone(),
+                }));
                 dirty = true;
```

Comment:
Is dropping this handling intentional?

Reply:
Yes, as we're now only passing the `Version`, instead of the `NodeVersion`, so we aren't passing the version of `npm`. Also, looking forward to the support of `volta install npm`, we don't want to write the `npm` version unless the user explicitly calls that command, otherwise we want to leave it alone.

This might be premature, however, so it may be better to pass both along and keep the behavior the same for now.